File: Easy/Longest_Common_Prefix/prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        out = ''
        fl = True
        
        strs.sort(key=lambda x: len(x))

        for idx, letter in enumerate(strs[0]):
            for word in strs[1:]:
                if letter != word[idx]:
                    fl = False 
            if fl:
                out += letter
            else:
                break
        return out

File: Easy/Longest_Common_Prefix/prefix.go
package main

func longestCommonPrefix(strs []string) string {
    fl := true 
    result := ""

    for idx, letter := range strs[0]{
        for _, word := range strs[1:] {
            if len(word) >= idx + 1 {
                if string(letter) != string(word[idx]){
                    fl = false
                }
            }else{
                fl = false
            }
        }
        if fl == true {
            result += string(letter)
        }else{
            break
        }
    }
    return result
}